day12/writex.py

```diff
@@ -0,0 +1,51 @@
+def writeX(matrix):
+    rows = set()
+    cols = set()
+
+    height = len(matrix[0])
+    width = len(matrix)
+
+    for i in xrange(height):
+        for j in xrange(width):
+            if matrix[i][j] == 'x':
+                rows.add(i)
+                cols.add(j)
+
+    for row in rows:
+        for j in xrange(width):
+            if matrix[row][j] != 'x':
```

Comment:
I feel like you don't need this if statement. And I feel like it's actually faster to just do `matrix[row][j] = 'x'`, because it takes fewer machine instructions than to load something into memory and then branch